packages/cohort_banking/src/app.rs

```diff
@@ -0,0 +1,136 @@
+        let counter_commits = meter.u64_counter("metric_commits").with_unit(Unit::new("tx")).init();
+        let counter_oo_installs = meter.u64_counter("metric_oo_installs").with_unit(Unit::new("tx")).init();
+        let counter_oo_no_data_found = meter.u64_counter("metric_oo_no_data_found").with_unit(Unit::new("tx")).init();
+
+        Ok(BankingApp {
+            config: config.clone(),
+            cohort_api: None,
+            database: Database::init_db(db_config).await.map_err(|e| e.to_string())?,
+            counter_aborts: Arc::new(counter_aborts),
+            counter_commits: Arc::new(counter_commits),
+            counter_oo_installs: Arc::new(counter_oo_installs),
+            counter_oo_no_data_found: Arc::new(counter_oo_no_data_found),
+        })
+    }
+
+    pub async fn init(&mut self) -> Result<(), String> {
+        let installer = StatemapInstallerImpl {
+            database: Arc::clone(&self.database),
+        };
+
+        let cohort_api = Cohort::create(self.config.clone(), installer).await.map_err(|e| e.to_string())?;
+
+        self.cohort_api = Some(cohort_api);
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Handler<TransferRequest> for BankingApp {
+    async fn handle(&self, request: TransferRequest) -> Result<(), String> {
+        log::debug!("processig new banking transfer request: {:?}", request);
+
+        let request_copy = request.clone();
+
+        let statemap = vec![HashMap::from([(
+            BusinessActionType::TRANSFER.to_string(),
+            TransferRequest::new(request.from.clone(), request.to.clone(), request.amount).json(),
+        )])];
+
+        let request = CertificationRequest {
+            timeout_ms: 0,
+            candidate: CandidateData {
+                readset: vec![request.from.clone(), request.to.clone()],
+                writeset: vec![request.from, request.to],
+                statemap: Some(statemap),
+            },
+        };
+
+        let state_provider = StateProviderImpl {
+            database: Arc::clone(&self.database),
+            request: request_copy.clone(),
+        };
+
+        let oo_inst = OutOfOrderInstallerImpl {
+            database: Arc::clone(&self.database),
+            request: request_copy,
+            detailed_logging: false,
+            counter_oo_installs: Arc::clone(&self.counter_oo_installs),
+            counter_oo_no_data_found: Arc::clone(&self.counter_oo_no_data_found),
+        };
+
+        match self
+            .cohort_api
+            .as_ref()
+            .expect("Banking app is not initialised")
+            .certify(request, &state_provider, &oo_inst)
+            .await
+        {
+            Ok(rsp) => {
+                let ca = Arc::clone(&self.counter_aborts);
+                let cc = Arc::clone(&self.counter_commits);
+                let is_abort = rsp.decision == Decision::Aborted;
+                let _ = tokio::spawn(async move {
```

Comment:
Same as mentioned in earlier comment, don't await on the `tasks` for capturing the opentelemetry metrics. This can run in the background and will improve the overall performance.

Reply:
Even if I miss to mention in other parts of the code, don't await on any of the `tasks` for adding data into open tel.